casper/src/main/scala/coop/rchain/casper/engine/BlockApproverProtocol.scala

```diff
@@ -178,11 +177,31 @@ object BlockApproverProtocol {
         _ <- (blockDeploys.size == genesisBlessedContracts.size)
               .either(())
               .or("Mismatch between number of candidate deploys and expected number of deploys.")
+
+        // check if expected blessed contracts and ones reported by candidate are equal
+        (leftover, _) = genesisBlessedContracts.toList.foldLeft(
+          (genesisBlessedContracts, false)
+        ) {
+          case ((leftover, inconsistencyFound), blessedDeploy) =>
+            if (inconsistencyFound) (leftover, inconsistencyFound)
+            else {
+              blockDeploys.find(_.deploy.data.term == blessedDeploy.data.term) match {
+                case Some(_) =>
+                  (leftover - blessedDeploy, false)
+                case None =>
+                  (leftover, true)
+              }
+            }
+        }
```

Comment:
Order should be also equal as from genesis creation.
https://github.com/rchain/rchain/blob/6507f3cf94859ef906eb0b0a4dd7f566babe14e5/casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala#L62-L74

Just for fun you can use fs2, zip it in a stream, map in parallel for equality and print to log top three unequal with 50 characters from deploy term.